presidio-analyzer/presidio_analyzer/analyzer_engine_provider.py

```diff
@@ -0,0 +1,132 @@
+import yaml
+import os
+import logging
+from pathlib import Path
+from typing import Optional, Union, List, Tuple, Dict
+
+from presidio_analyzer import AnalyzerEngine, RecognizerRegistry, EntityRecognizer, PatternRecognizer
+from presidio_analyzer.nlp_engine import NlpEngineProvider, NlpEngine
+import presidio_analyzer.recognizer_registry
+
+logger = logging.getLogger("presidio-analyzer")
+
+
+class AnalyzerEngineProvider:
+    """
+    Utility function for loading Presidio Analyzer.
+
+    Use this class to load presidio analyzer engine from a yaml file
+    """
+
+    def __init__(self, conf_file: Optional[Union[Path, str]] = None):
+        self.configuration = {}
+        if not os.path.exists(conf_file):
+            logger.warning(
+                "configuration file is missing. Using default configuration for analyzer engine"
+            )
+            return
+        
+        self.configuration = yaml.safe_load(open(conf_file))
+        return
+
+
+    def create_engine(self) -> AnalyzerEngine:
+        """
+        loads Presidio Analyzer from yaml configuration file.
+
+        :return: analyzer engine initialized with yaml configuration
+        """
+
+        nlp_engine = self._load_nlp_engine()
+        registry = self._load_recognizer_registry()
+        supported_languages = self.configuration.get("supported_languages", None)
+        default_score_threshold = self.configuration.get("default_score_threshold", 0)
+
+        analyzer = AnalyzerEngine(
+            nlp_engine=nlp_engine, 
+            registry=registry,
+            supported_languages=supported_languages,
+            default_score_threshold=default_score_threshold
+        )
+
+        return analyzer
+    
+    def _is_enabled(self, recognizer: Dict) -> bool:
+        return not "enabled" in recognizer or recognizer["enabled"]
+
+    def _get_name(self, recognizer) -> str:
+        if isinstance(recognizer, str):
+            return recognizer
+        return recognizer["name"]
+
+    def _get_languages(self, recognizer) -> List[str]:
+        if isinstance(recognizer, str):
+            return ["en"]
+        return recognizer["supported_languages"]
+
+    def _split_recognizers(self, recognizers) -> Tuple[List[str],List[str]]:
+        """
+        splits the recognizer list to predefined and custom
+        """
+        predefined =[]
+        custom =[]
+
+        predefined = [recognizer for recognizer in recognizers if isinstance(recognizer, str) or ("type" in recognizer and recognizer["type"] == "predefined")]
+        # all recognizers are custom by default, type: custom can be mentioned as well
+        custom = [recognizer for recognizer in recognizers if not isinstance(recognizer, str) and ("type" not in recognizer or recognizer["type"] == "custom")]
+        return predefined, custom
+
+    def _create_recognizers(self, recognizer: Dict) -> List[EntityRecognizer]:
```

Comment:
```suggestion
    def _create_custom_recognizers(self, recognizer: Dict) -> List[EntityRecognizer]:
```
nit: differentiate between the creation of predefined and custom